lib/galaxy/jobs/command_factory.py

```diff
@@ -244,6 +244,8 @@ def __handle_metadata(commands_builder, job_wrapper, runner, remote_command_para
 
 def __copy_if_exists_command(work_dir_output):
     source_file, destination = work_dir_output
+    if '?' in source_file or '*' in source_file:
+        source_file = source_file.replace('*', '"*"').replace('?', '"?"')
```

Comment:
Clever - thanks for cleaning this up.

Reply:
You could do this unconditionally, saves one or two sequential scans, but given the tiny size of the data it doesn't matter. So, +1.

Reply:
@ic4f that wouldn't be very pythonic I think 😈

Reply:
@martenson Why not? Both `a in b` and `str.replace` are O(n). In the worst case (str contains `*` but not `?`) you get 4 linear scans if you use the conditional, but without it the worst case is 2 scans. Besides, isn't "pythonic" simplicity and elegance, and also EAFP?

Reply:
@ic4f this is obviously very subjective from me but for me "explicit is better then implicit" is the golden rule.

Nevertheless my comment was more of a joke, please disregard it. Your approach is elegant.

Reply:
If it were not for github, where else would be argue about the many interpretations of the Zen of Python? :rofl:

Reply:
I mean, you are utterly wrong about that piece of code, and everybody knows it...😆